frontend/src/types.ts

```diff
@@ -277,12 +277,42 @@ export interface ToolbarProps extends EditorProps {
 
 export type PropertyFilterValue = string | number | (string | number)[] | null
 
+// duplicated from plugin server. Follow-up to de-duplicate
+export enum UnixTimestampPropertyTypeFormat {
+    UNIX_TIMESTAMP = 'unix_timestamp',
+    UNIX_TIMESTAMP_MILLISECONDS = 'unix_timestamp_milliseconds',
+}
+
+// duplicated from plugin server. Follow-up to de-duplicate
+export enum DateTimePropertyTypeFormat {
+    ISO8601_DATE = 'YYYY-MM-DDThh:mm:ssZ',
+    FULL_DATE = 'YYYY-MM-DD hh:mm:ss',
+    FULL_DATE_INCREASING = 'DD-MM-YYYY hh:mm:ss',
+    DATE = 'YYYY-MM-DD',
+    RFC_822 = 'rfc_822',
+    WITH_SLASHES = 'YYYY/MM/DD hh:mm:ss',
+    WITH_SLASHES_INCREASING = 'DD/MM/YYYY hh:mm:ss',
+}
+
+// duplicated from plugin server. Follow-up to de-duplicate
+export type PropertyTypeFormat = DateTimePropertyTypeFormat | UnixTimestampPropertyTypeFormat
+
+// duplicated from plugin server. Follow-up to de-duplicate
+export enum PropertyType {
+    DateTime = 'DateTime',
+    String = 'String',
+    Numeric = 'Numeric',
+    Boolean = 'Boolean',
+}
+
 export interface PropertyFilter {
     key: string
     operator: PropertyOperator | null
     type: string
     value: PropertyFilterValue
     group_type_index?: number | null
+    property_type?: PropertyType | undefined
+    property_type_format?: PropertyTypeFormat | undefined
```

Comment:
We use this type liberally everywhere, including in the URLs, and on insights/dashboard-items in the database. These two new variables hint that this information, when existing, will be automatically available on all filters, everywhere... And that's actually not going to be the case.

Just like we don't store the name of a cohort in a filter (just the ID and then load the name async), is this also really needed, and could we avoid this?

Reply:
I was passing it via the API to avoid the lookup on the backend (assuming I've understood the prompt)

But it is more shared state. Removing that feels like a follow-up to this PR, rather than stretching this one out...

Reply:
I'd much rather have a lookup on the backend. It's safer, and doesn't introduce another place where data can go out of sync. Imagine editing the property definition and changing something in your datetime's format. Do you then need to go through every dashboard that has an insight with this property, and recreate it?

Also, can't we infer the `property_type`, and discard `_format`? Looking at this `afterMount` addition:

```
{
    key: '$time',
    operator: PropertyOperator.IsDateAfter,
    type: 'event',
    // clickhouse can't (by default) parse an ISO-8601 DATE in this position
    value: now().subtract(values.months, 'months').format('YYYY-MM-DD hh:mm:ss'),
    property_type: PropertyType.DateTime,
    property_type_format: DateTimePropertyTypeFormat.FULL_DATE,
},
```

If it's a date can be inferred from the operator, and do we really need to also store the format?

There are so many places that deal with filters, and we probably have tens or hundreds of thousands of them in the wild. Making a quick addition like that really tickles my "there are too many unknown unknowns for me to feel comfortable with this" sense, and I fear we're underestimating the scope here.